docs/move/move-on-aptos/objects/using-objects.md

```diff
@@ -0,0 +1,181 @@
+---
+title: "Using objects"
+---
+
+# Using objects
+
+## Using an object as an entry function argument
+
+`0x1::object::Object<T>` can be used as an input argument to an entry function.
+It's encoded as the `address` of the object, and type checked to the value
+in `T`.
+
+For more generic functions, this can be handled as providing a generic `T` for
+typing the input:
+
+```move
+module my_addr::object_playground {
+  use aptos_framework::object::Object;
+
+  /// This will fail of the object doesn't have the generic `T` stored
+  entry fun do_something<T>(object: Object<T>) {
+    // ...
+  }
+}
+```
+
+Additionally, the individual resources that are at the object can be used. The
+resource will be checked for existence before the function executes.
+
+```move
+module my_addr::object_playground {
+  use aptos_framework::object::Object;
+
+  struct MyAwesomeStruct has key {}
+
+  /// This will fail if the object doesn't have MyAwesomeStruct stored
+  entry fun do_something(object: Object<MyAwesomeStruct>) {
+    // ...
+  }
+}
+```
+
+The resource `0x1::object::ObjectCore` is available for all objects
+
+```move
+module my_addr::object_playground {
+  use aptos_framework::object::Object;
+
+  /// This will only fail of the address is not an object
+  entry fun do_something(object: Object<Object>) {
+    // ...
+  }
+}
+```
+
+# Object Types
+
+Objects can store multiple resources at their address, and you can refer to an
+Object by any of those types. You can convert an address to an object, or
+convert
+an object between types as long as the resource is available.
+
+```move
+module my_addr::object_playground {
+  use aptos_framework::object::{self, Object, ObjectCore};
+
+  struct MyAwesomeStruct has key {}
+
+  fun convert_type(object: Object<ObjectCore>): Object<MyAwesomeStruct> {
+    object::convert<MyAwesomeStruct>(object)
+  }
+
+  fun address_to_type(object_address: address): Object<MyAwesomeStruct> {
+    object::address_to_object<MyAwesomeStruct>(object)
+  }
+}
+```
+
+# Object ownership
+
+Objects can be owned by any address, this includes Objects, Accounts, and
+Resource
+accounts. This allows composability between objects, as well as complex
+relationships
+between objects.
+
+## Looking up ownership
+
+Ownership can be found by a few methods:
+
+```move
+module my_addr::object_playground {
+  use aptos_framework::object::{self, Object};
+
+  fun lookup_ownership<T>(object: Object<T>) {
+    object::is_owner(object, @my_addr); // Returns true if it's the owner
+    object::owner(object); // Returns the owner's address
```

Comment:
This would better be expressed as useful code, perhaps some variant on:
```
    let owner = object::owner(object);            // object::owner returns owner's address
    assert(object::is_owner(object, owner));  // object:is_owner returns true if it's the owner
    let _i_own_it = object::is_owner(object, @my_addr);
```